packages/@expo/cli/src/start/server/middleware/CorsMiddleware.ts

```diff
@@ -33,7 +33,14 @@ export function createCorsMiddleware(exp: ExpoConfig) {
         return;
       }
 
-      res.setHeader('Access-Control-Allow-Origin', req.headers.origin);
+      // Skip for localhost since we shouldn't allow escalating cross-origin requests
+      // across separate sites running on localhost. If the `Origin` is set to a localhost
+      // hostname, the browser is either likely to disregard CORS anyway, or we're not
+      // making a cross-origin request anyway
```

Comment:
From my research browsers treat origins with a localhost hostname the same as other origins in the contexts of origin policies. (localhost is special-cased for unrelated things like secure contexts)
```suggestion
      // Skip for localhost since we shouldn't allow escalating cross-origin requests
      // across separate sites running on localhost. If the `Origin` is set to a localhost
      // hostname, we're not making a cross-origin request anyway
```